packages/kbn-expandable-flyout/src/context/memory_state_provider.tsx

```diff
@@ -6,19 +6,54 @@
  * Side Public License, v 1.
  */
 
-import React, { FC, PropsWithChildren, useCallback, useMemo, useReducer } from 'react';
-import { ActionType } from '../actions';
+import React, { createContext, FC, useCallback, useMemo } from 'react';
+import {
+  createDispatchHook,
+  createSelectorHook,
+  Provider as ReduxProvider,
+  ReactReduxContextValue,
+} from 'react-redux';
+import { configureStore } from '@reduxjs/toolkit';
+
 import { reducer } from '../reducer';
-import type { ExpandableFlyoutContextValue, FlyoutPanelProps } from '../types';
-import { initialState } from '../reducer';
-import { ExpandableFlyoutContext } from '../context';
+import { State } from '../state';
+import type { ExpandableFlyoutApi, FlyoutPanelProps } from '../types';
+import {
+  closeLeftPanelAction,
+  closePanelsAction,
+  closePreviewPanelAction,
+  closeRightPanelAction,
+  openLeftPanelAction,
+  openPanelsAction,
+  openPreviewPanelAction,
+  openRightPanelAction,
+  previousPreviewPanelAction,
+} from '../actions';
 
-/**
- * In-memory state provider for the expandable flyout, for cases when we don't want changes to be persisted
- * in the url.
- */
-export const MemoryStateProvider: FC<PropsWithChildren<{}>> = ({ children }) => {
-  const [state, dispatch] = useReducer(reducer, initialState);
+const initialState: State = {
+  left: undefined,
+  right: undefined,
+  preview: [],
+};
```

Comment:
any reason for not using the already existing `initialState` constant in the `state.ts` file?